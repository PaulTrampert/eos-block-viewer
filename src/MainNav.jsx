import React from 'react';
import PropTypes from 'prop-types';
import Navbar from 'react-bootstrap/Navbar';
import Button from 'react-bootstrap/Button';
import {ReactComponent as EosIcon} from './EOS-icon.svg';

function MainNav({onRequestBlocks, loadingBlocks}) {
  return (
    <Navbar bg="dark" variant="dark" className="justify-content-between">
      <Navbar.Brand>
        <EosIcon style={{marginRight: '0.5em'}} />
        EOS Block Viewer
      </Navbar.Brand>
      <Button onClick={onRequestBlocks} disabled={loadingBlocks}>Load</Button>
    </Navbar>
  );
}

MainNav.propTypes = {
  onRequestBlocks: PropTypes.func.isRequired
}

export default MainNav;