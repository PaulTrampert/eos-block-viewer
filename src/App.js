import React from 'react';
import MainNav from './MainNav';
import api from './apiWrapper';
import BlockTable from './BlockTable';

class App extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      loadingBlocks: false,
      blocks: []
    };
  }

  loadBlocks = async () => {
    this.setState({
      loadingBlocks: true
    });

    const blocks = await api.getLastBlocks();

    this.setState({
      loadingBlocks: false,
      blocks
    });
  }

  render = () => {
    let {
      blocks,
      loadingBlocks
    } = this.state;

    return (
      <>
        <MainNav onRequestBlocks={this.loadBlocks} loadingBlocks={loadingBlocks} />
        <BlockTable blocks={blocks} />
      </>
    );
  }
}

export default App;
