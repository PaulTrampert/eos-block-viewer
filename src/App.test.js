import React from 'react';
import {shallow} from 'enzyme';
import App from './App';
import MainNav from './MainNav';
import api from './apiWrapper';

jest.mock('./apiWrapper');

describe('App', () => {
  let subject;
  let instance;

  beforeEach(() => {
    subject = shallow(<App />);
    instance = subject.instance();
  })

  describe('loadBlocks', () => {
    let blocks;

    beforeEach(() => {
      blocks = [];
      api.getLastBlocks.mockReturnValue(blocks);
    });

    it('loads blocks from the api', async () => {
      await instance.loadBlocks();

      expect(api.getLastBlocks).toHaveBeenCalledWith();
    });

    it('puts the blocks in state', async () => {
      await instance.loadBlocks();

      expect(subject.state().blocks).toBe(blocks);
    });
  });

  describe('render', () => {
    it('renders the main nav and block table', () => {
      expect(subject).toMatchSnapshot();
    });

    it("binds loadBlocks to MainNav's onRequestBlocks event", () => {
      expect(subject.find(MainNav).props().onRequestBlocks).toBe(subject.instance().loadBlocks);
    });

    describe('while blocks are loading', () => {
      beforeEach(() => {
        subject.setState({loadingBlocks: true});
      });

      it('passes the loadingBlocks state to the MainNav', () => {
        expect(subject).toMatchSnapshot();
      });
    });
  });
});
