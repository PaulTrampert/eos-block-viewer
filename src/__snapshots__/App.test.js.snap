// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render renders the main nav and block table 1`] = `
<Fragment>
  <MainNav
    loadingBlocks={false}
    onRequestBlocks={[Function]}
  />
  <BlockTable
    blocks={Array []}
    loadingBlocks={false}
  />
</Fragment>
`;

exports[`App render while blocks are loading passes the loadingBlocks state to the MainNav and BlockTable 1`] = `
<Fragment>
  <MainNav
    loadingBlocks={true}
    onRequestBlocks={[Function]}
  />
  <BlockTable
    blocks={Array []}
    loadingBlocks={true}
  />
</Fragment>
`;
